.circleci/README.md

```diff
@@ -0,0 +1,100 @@
+## CircleCI
+
+#### config.yml overview:
+
+##### Pipeline parameters:
+    - workbench-image
+    - db-image
+    - circle-service-account
+    
+##### Anchors
+    - default-environment
+    - db-environment
+    - puppeteer-test-environment
+    - filter-master-branch
+    - filter-release-tags
+    - filter-pr-branch
+
+##### Executors
+    - workbench-executor
+    - db-executor
+    - e2e-test-executor
+    
+##### Commands
+    - notify-on-failure
+    - gcloud-auth-login
+    - checkout-code
+    - store-commit-message
+    - halt-e2e-test-check
+    - halt-test-check
+    - ui-cache
+    - run-e2e-test
+    - launch-local-api
+    - launch-local-ui
+    - run-api-test
+    - deploy-to-staging-perf
+    - deploy-to-test
+    
+##### Jobs
```

Comment:
[not necessarily for this PR]

Thinking about naming here, a better convention might be for each job to start with a verb. That way they sort a little better, and align with typical method-naming conventions where you want the verb first. This is also the pattern that most of the commands use.

A few example renames:
  api-unit-test --> test-api-unit
  api-deps-check --> check-api-deps
  ui-deploy-to-test --> deploy-ui-to-test
etc...

This may be infeasible / annoying to change though, with not great value add. Just thought it was worth noting.